lib/pavilion/commands/log.py

```diff
@@ -129,9 +137,58 @@ def run(self, pav_cfg, args):
             file_name = test.path/self.LOG_PATHS[cmd_name]
 
         if not file_name.exists():
-            output.fprint(self.errfile, "Log file does not exist: {}"
-                          .format(file_name), color=output.RED)
-            return 1
+            if args.follow:
+                if 'build' in cmd_name:
+                    file_name = Path(pav_cfg.working_dir, 'test_runs/', args.id,
+                                'build/pav_build_log')
+                    build_hash_log = ''
+
+                    if not file_name.exists():
+                        status_file = Path(pav_cfg.working_dir, 'test_runs/', args.id, 'status')
+                        with status_file.open() as file:
+                            for line in file:
+                                if 'BUILDING Starting build' in line:
+                                    *_, build_hash = line.split()
+                                    build_hash = build_hash[:-1]
+                                    hash_log = '{}.log'.format(build_hash)
+                                    build_hash_log = Path(pav_cfg.working_dir, 'builds/',
+                                                     hash_log)
+                        
+                        while not build_hash_log.exists():
+                            output.fprint(self.outfile, 'hash log doesn\'t exist. Checking again...', 
+                            end='\r', flush=True)
+                            if file_name.exists():
+                                break
+                            time.sleep(.5)
+
+                        with build_hash_log.open() as file:
+                            for line in file:
+                                output.fprint(self.outfile, line, width=None, end='', flush=True)
+                            file.seek(0,2)
+                            while not file_name.exists():
+                                line = file.readline()
+                                if line:
+                                    output.fprint(self.outfile, line, width=None, end='', flush=True)
+                                else: 
+                                    time.sleep(.5)
+
+                    with file_name.open() as file:
+                        file.seek(0,2)
+                        while True:
+                            data = file.read()
+                            if not data:
+                                time.sleep(.5)
+                            else:
+                                output.fprint(self.outfile, data, end='', flush=True)
```

Comment:
Instead of jumping through all these possibilities, use a list of possible filename and just check each of them in turn.